File: packages/dd-trace/test/llmobs/parity/scenarios/openai/chat-completion-error.js
'use strict'

const { finish, loadTracer, loadVersionedModule } = require('../common')

async function main () {
  const tracer = loadTracer('openai')
  const OpenAI = loadVersionedModule('openai')
  const client = new OpenAI({
    apiKey: process.env.OPENAI_API_KEY,
    baseURL: `${process.env.PROVIDER_BASE_URL}/v1`,
    maxRetries: 0,
  })
  try {
    await client.chat.completions.create({
      model: 'gpt-4o-mini',
      messages: [{ role: 'user', content: 'Say parity' }],
      max_tokens: 8,
    })
  } catch {}
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack ?? error}\n`)
  process.exitCode = 1
})

File: packages/dd-trace/test/llmobs/parity/scenarios/openai/chat-completion-error.py
import os

from openai import OpenAI

from common import finish, start


start("openai")
client = OpenAI(
    api_key=os.environ["OPENAI_API_KEY"],
    base_url=f'{os.environ["PROVIDER_BASE_URL"]}/v1',
    max_retries=0,
)
try:
    client.chat.completions.create(
        model="gpt-4o-mini",
        messages=[{"role": "user", "content": "Say parity"}],
        max_tokens=8,
    )
except Exception:
    pass
finish()
